### PR TITLE
Handle session present

### PR DESCRIPTION
**Pull request by nedimtokic**
If the session present is false, do send introspection and empty cache message.

Closes: #5

### CODE REVIEW

File: AstarteDeviceSDKCSharp/Transport/MQTT/AstarteMqttTransport.cs [hunk 1]
```diff
@@ -67,9 +67,9 @@ private async Task<IMqttClient> InitClientAsync()
...
         }
 
-        private async Task CompleteAstarteConnection()
+        private async Task CompleteAstarteConnection(bool IsSessionPresent)
         {
-            if (!_introspectionSent)
+            if (!IsSessionPresent || !_introspectionSent)
             {
                 await SetupSubscriptions();
                 await SendIntrospection();
```
Comment (harlem88) [inline AstarteDeviceSDKCSharp/Transport/MQTT/AstarteMqttTransport.cs:75]: For a complete `session present` handling, you need to resend all known properties from the device, so  you need to add the `ResendAllProperties` method here.

See see the java-sdk for reference: 
https://github.com/astarte-platform/astarte-device-sdk-java/blob/5b2f878dad8d455766ef209b81b5e1f101a438d8/DeviceSDK/src/main/java/org/astarteplatform/devicesdk/transport/mqtt/AstarteMqttV1Transport.java#L474

Comment (nedimtokic) [reply AstarteDeviceSDKCSharp/Transport/MQTT/AstarteMqttTransport.cs:75]: Added method for sending properties from local storage to Astarte when session present is true.

File: AstarteDeviceSDKCSharp/Transport/MQTT/AstarteMqttV1Transport.cs [hunk 2]
```diff
@@ -154,5 +154,42 @@ private int QosFromReliability(AstarteInterfaceDatastreamMapping mapping)
             }
         }
 
+        public override async Task ResendAllProperties()
+        {
+            if (_astartePropertyStorage.Equals(null))
```
Comment (harlem88) [inline AstarteDeviceSDKCSharp/Transport/MQTT/AstarteMqttV1Transport.cs:159]: Can `_astartePropertyStorage` be null? in AstarteMqttV1Transport base class was declare as `non-nullable` reference... Or am I getting it wrong?

**Comment by harlem88** [comment]
Also add a line to changelog :
`Handling sessionPresent flag  from the broker`

**Review by harlem88 — APPROVED** [review]

**Review by osmanhadzic — APPROVED** [review]

**Pull request merged**